solr/core/src/java/org/apache/solr/cli/CreateTool.java

```diff
@@ -43,24 +68,295 @@ public String getName() {
+                .getZkClient()
+                .exists("/configs/" + confName, true);
+
+    if (CollectionAdminParams.SYSTEM_COLL.equals(collectionName)) {
+      // do nothing
+    } else if (configExistsInZk) {
+      echo("Re-using existing configuration directory " + confName);
+    } else { // if (confdir != null && !confdir.trim().isEmpty()) {
+      if (confName == null || confName.trim().isEmpty()) {
+        confName = collectionName;
+      }
+      Path confPath = ConfigSetService.getConfigsetPath(confDir, configsetsDir);
+
+      echoIfVerbose(
+          "Uploading "
+              + confPath.toAbsolutePath()
+              + " for config "
+              + confName
+              + " to ZooKeeper at "
+              + cloudSolrClient.getClusterStateProvider().getQuorumHosts(),
+          cli);
+      ZkMaintenanceUtils.uploadToZK(
+          ZkStateReader.from(cloudSolrClient).getZkClient(),
+          confPath,
+          ZkMaintenanceUtils.CONFIGS_ZKNODE + "/" + confName,
+          ZkMaintenanceUtils.UPLOAD_FILENAME_EXCLUDE_PATTERN);
+    }
+
+    // since creating a collection is a heavy-weight operation, check for existence first
+    if (SolrCLI.safeCheckCollectionExists(solrUrl, collectionName)) {
+      throw new IllegalStateException(
+          "\nCollection '"
+              + collectionName
+              + "' already exists!\nChecked collection existence using CollectionAdminRequest");
+    }
+
+    // doesn't seem to exist ... try to create
+    echoIfVerbose(
+        "\nCreating new collection '" + collectionName + "' using CollectionAdminRequest", cli);
+
+    NamedList<Object> response;
+    try {
+      response =
+          cloudSolrClient.request(
+              CollectionAdminRequest.createCollection(
+                  collectionName, confName, numShards, replicationFactor));
+    } catch (SolrServerException sse) {
+      throw new Exception(
+          "Failed to create collection '" + collectionName + "' due to: " + sse.getMessage());
+    }
+
+    if (cli.hasOption(SolrCLI.OPTION_VERBOSE.getOpt())) {
+      CharArr arr = new CharArr();
+      new JSONWriter(arr, 2).write(response.asMap());
+      echo(arr.toString());
+    } else {
+      String endMessage =
+          String.format(
+              Locale.ROOT,
+              "Created collection '%s' with %d shard(s), %d replica(s)",
+              collectionName,
+              numShards,
+              replicationFactor);
+      if (confName != null && !confName.trim().isEmpty()) {
+        endMessage += String.format(Locale.ROOT, " with config-set '%s'", confName);
+      }
+
+      echo(endMessage);
+    }
+  }
+
+  private Path getFullConfDir(Path confDirName, Path solrInstallDir) {
+    Path configSetsPath = Paths.get("server/solr/configsets/");
+    return solrInstallDir.resolve(configSetsPath).resolve(confDirName);
+  }
+
+  private void ensureConfDirExists(Path confDirName, Path solrInstallDir) {
```

Comment:
nitpick but do consider parameter order.  I think solrInstallDir should come first on this method and the previous method because confDirName is surely beneath it

Reply:
Working on this....